src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -691,11 +691,16 @@ def __ensure_error_message_restriction_compliance(full_message):
     @staticmethod
     def __add_error(add_to, detail):
         """ Add formatted error object to given errors list """
+        for error_detail in add_to:
+            if detail["message"] == error_detail["message"]:
+                return False
+
```

Comment:
Consider the case where this message [A] is already in the list, and message [B] comes along. Can you remove [A] from the list and replace it with the more "expressive" message [B] too?

[A] -- "I/O exception copying file"
[B] -- "Execution stopped due to failure. Exception: I/O exception copying file."

Having A in there while B gets reported next is sort of redundant.

This will mean you will also additionally need to to see if any entry in the list is also contained within the new message that is coming in.

Reply:
Also, note, when this is done, the reverse chronological order of message arrival should be preserved. Newest message is first in the list.

Reply:
Good idea. Added logic to do this.

One thing that we might want to consider is the error codes of the messages - should they also be required to match when determining whether to add a message or not? I made the assumption that if the error message is similar enough (contains it), then the error is probably on the same severity level.

Reply:
The updated implementation doesn't have the newest arriving message as first in the list if there is a message in can replace lower in the list. (Messages are reverse chronological.. )

B' => [X, A, B, C]   -- becomes -- > [B', X, A, C],  not [X, A, B', C]

Another approach, which is probably better is to do it only for the current first element. Less verbose messages may remain if they are older than the stored newest.

Reply:
Adjusted code to preserve reverse chronological ordering of error messages.